docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,805 @@ sidebar: lb4_sidebar
+    "content": {
+      "type": "string"
+    }
+  },
+
+  "mixins": {
+    "FindByTitle": {
+      "returnArgumentName": "notes"
+    },
+
+    "Category": true
+  }
+}
+```
+
+Specifying an options object for `FindByTitle` is the same as specifying a value
+of **true** as it will apply the **findByTitle.js** custom model method/remote
+method mixin to the `Note` model. A value of **false** will not apply the mixin.
+
+## LoopBack 4 Approach to Creating A Custom Model Method Mixin And Remote Model Method Mixin
+
+As mentioned in the previous section, the
+[Add a New Model Method And a New Endpoint](./methods.md#add-a-new-model-method-and-a-new-endpoint)
+section of the [Migrating custom model methods](./methods.md) document explains
+how a LoopBack 3 developer can define a custom model method named `findByTitle`
+on the `Note` model, and define a remote method to make it available as a new
+endpoint. It then shows how a LoopBack 4 developer can implement a `findByTitle`
+method on the `NoteRepository` and on the `NoteController` to accomplish the
+same thing.
+
+In this section, we will show how a LoopBack 4 developer can define two mixins (
+a repository mixin and a controller mixin) to add a `findByTitle` method to
+`NoteRepository` and `NoteController` respectively.
+
+In LoopBack 4, a developer is able to create a repository and controller method
+mixin by:
+
+- defining a common interface for both mixin class factory functions
+- placing the mixin class factory functions in separate files
+- generating a repository and controller using the CLI as usual
+- adjusting the repository and controller files to make use of its respective
+  mixin class factory function
+
+### Defining A Common Interface For The findByTitle Method
+
+Let's define a common interface `FindByTitleInterface` in
+**src/mixins/findByTitleInterface.ts**.
+
+{% include code-caption.html content="src/mixins/findByTitleInterface.ts" %}
+
+```ts
+import {Model} from '@loopback/repository';
+
+export interface FindByTitleInterface<M extends Model> {
+  findByTitle(title: string): Promise<M[]>;
+}
+```
+
+### Defining A Repository Mixin Class Factory Function
+
+In **src/mixins/findByTitleRepositoryMixin.ts**, let's define the mixin class
+factory function `FindByTitleRepositoryMixin` which adds the `findByTitle`
+method to any repository.
+
+{% include code-caption.html content="src/mixins/findByTitleRepositoryMixin.ts" %}
+
+```ts
+import {Constructor} from '@loopback/context';
+
+import {Model} from '@loopback/repository';
+import {FindByTitleInterface} from './findByTitleInterface';
+
+export function FindByTitleRepositoryMixin<
+  M extends Model,
+  T extends Constructor<any>
+>(superClass: T) {
+  return class extends superClass implements FindByTitleInterface<M> {
+    async findByTitle(title: string): Promise<M[]> {
+      const titleFilter = {
+        where: {
+          title: title,
+        },
+      };
+      const foundItems = await this.find(titleFilter);
```

Comment:
a question...if the mixin doesn't have a function `find`, how could it call `find` here?

Reply:
@jannyHou `find` is a function available on all models in lb3.

Reply:
oh I didn't mean lb3, I mean, typescript doesn't complain sth like `find` not found?

Reply:
no transpilation errors.

Reply:
@raymondfeng , any idea why there is no complaint of `this.find()`?

I know this code works:

```
export function FindByTitleRepositoryMixin<
  M extends Model,
  R extends Constructor<any>
>(superClass: R) {
  return class extends superClass implements FindByTitleInterface<M> {
    async findByTitle(title: string): Promise<M[]> {
      const titleFilter = {
        where: {
          title: title,
        },
      };
      return await this.find(titleFilter);
    }
  };
}
```
But just curious why ts doesn't complain.

Reply:
Because `R` is defined as `Constructor<any>`, your super class is effectively extending from `any`. If you hover your mouse over `this.find` in VS Code, you will see that it has `any` type.

<img width="621" alt="Screen Shot 2020-02-27 at 14 34 46" src="https://user-images.githubusercontent.com/1140553/75450007-52e16800-596e-11ea-8a6f-c6b73a8d8121.png">

Here is a more correct definition of `FindByTitleRepositoryMixin`:

```ts
export function FindByTitleRepositoryMixin<
  M extends Model,
  R extends Constructor<CrudRepository<M>>
>(superClass: R) {
  return class extends superClass implements FindByTitleInterface<M> {
    // ...
  };
}
```

I am choosing `CrudRepository` because that's the interface where `find` is defined. By using that interface instead of e.g. `DefaultCrudRepository`, we make it possible to use this mixin with other repository implementations too.

Reply:
The problem with the code above is that TS now complains that `title` cannot be used in a filter, because a generic `Model` does not necessarily have a filter property. The solution is to limit `M` to `extends Model & {title: string}`.

```ts
export function FindByTitleRepositoryMixin<
  M extends Model & {title: string},
  R extends Constructor<CrudRepository<M>>
>(superClass: R) {
```

Unfortunately that's still not enough, because our `Condition<MT>` type cannot correctly handle the case where `MT` is a generic argument extending `Model & {title: string}`. I am proposing to use the following implementation:

```ts
    async findByTitle(title: string): Promise<M[]> {
      const where = {title} as Where<M>;
      const titleFilter = {where};
      return this.find(titleFilter);
    }
```

/cc @raymondfeng would you like to take a look on why `{title}` cannot be assigned to `Condition<M>` where `M` is a generic argument defined as `M extends Model & {title: string}`?

Example code to reproduce:

```ts
function test<M extends Model & {title: string}>(): Condition<M> {
  return {title: 'foo'};
}
```

Compiler error:

```
Type '{ title: "foo"; }' is not assignable to type 'Condition<M>'.ts(2322)
```